code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -21,16 +21,17 @@
 			if(M && istype(M) && !M.stat && M.client && (!hivenumber || M.hivenumber == hivenumber)) //Only living and connected xenos
 				to_chat(M, SPAN_XENODANGER("<span class=\"[fontsize_style]\"> [message]</span>"))
 
-//Sends a maptext alert to our currently selected squad. Does not make sound.
+//Sends a maptext alert to xenos.
 /proc/xeno_maptext(text = "", title_text = "", hivenumber = XENO_HIVE_NORMAL)
 	if(text == "" || !hivenumber)
 		return //Logic
 
 	if(SSticker.mode && SSticker.mode.xenomorphs.len) //Send to only xenos in our gamemode list. This is faster than scanning all mobs
-		for(var/datum/mind/L in SSticker.mode.xenomorphs)
-			var/mob/living/carbon/M = L.current
-			if(M && istype(M) && !M.stat && M.client && M.ally_of_hivenumber(hivenumber)) //Only living and connected xenos
-				M.play_screen_text("<span class='langchat' style=font-size:16pt;text-align:center valign='top'><u>[title_text]</u></span><br>" + text, /atom/movable/screen/text/screen_text/command_order, "#b491c8")
+		for(var/datum/mind/living in SSticker.mode.xenomorphs)
+			var/mob/living/carbon/xeno = living.current
+			if(xeno && istype(xeno) && !xeno.stat && xeno.client && xeno.hivenumber == hivenumber) //Only living and connected xenos
```

Comment:
If your change here from ally_of_hivenumber to specifically the hivenumber fixes the issue where foxtrot sometimes get xeno hive orders, then the faction of foxtrot is probably messed up.

Reply:
first time I hear about this bug, really funny.

Reply:
why foxtrot marines would be in SSticker.mode.xenomorphs tho... Unless they played xeno previously? I can change `var/mob/living/carbon/xeno` to `var/mob/living/carbon/xenomorph/xeno` so typecheck would make sure no marines get the message.

Reply:
My comment isn't a requested change, but I am noting that your changed from ally to hivenumber and mentioning an unrelated bug that *might* get fixed because of your change (but reproduction of it has not been reliable).